linetools/isgm/abscomponent.py

```diff
@@ -108,30 +109,37 @@ def from_component(cls, component, **kwargs):
                    A=component.A, name=component.name, **kwargs)
 
     @classmethod
-    def from_dict(cls, idict, **kwargs):
+    def from_dict(cls, idict, coord=None, **kwargs):
         """ Instantiate from a dict
 
         Parameters
         ----------
         idict : dict
+        use_line_list : str, optional
+          Name of the Linelist to use when generating the SpectralLines
 
         Returns
         -------
 
         """
-        slf = cls(SkyCoord(ra=idict['RA']*u.deg, dec=idict['DEC']*u.deg),
-                  tuple(idict['Zion']), idict['zcomp'], idict['vlim']*u.km/u.s,
+        if coord is not None:
+            radec = coord
+        else:
+            radec = SkyCoord(ra=idict['RA']*u.deg, dec=idict['DEC']*u.deg)
+        # Init
+        #slf = cls(radec, tuple(idict['Zion']), idict['zcomp'], Quantity(idict['vlim'], unit='km/s'),
+        slf = cls(radec, tuple(idict['Zion']), idict['zcomp'], idict['vlim']*u.km/u.s,
                   Ej=idict['Ej']/u.cm, A=idict['A'],
                   Ntup = tuple([idict[key] for key in ['flag_N', 'logN', 'sig_logN']]),
                   comment=idict['comment'], name=idict['Name'])
         # Add lines
         for key in idict['lines'].keys():
-            iline = SpectralLine.from_dict(idict['lines'][key])
+            iline = SpectralLine.from_dict(idict['lines'][key], coord=coord, **kwargs)
             slf.add_absline(iline, **kwargs)
         # Return
         return slf
 
-    def __init__(self, radec, Zion, z, vlim, Ej=0./u.cm, A=None,
+    def __init__(self, radec, Zion, z, vlim, Ej=Quantity(0., unit='1/cm'), A=None,
```

Comment:
Is there an advantage by using `Ej=Quantity(0., unit='1/cm')` instead of `Ej=0./u.cm`?